spec/spec_helper.cr

```diff
@@ -77,6 +77,24 @@ def semantic(node : ASTNode, *, warnings = nil, wants_doc = false, flags = nil)
   SemanticResult.new(program, node)
 end
 
+record TopLevelSemanticResult,
+  program : Program,
+  node : ASTNode
+
+def top_level_semantic(code : String, wants_doc = true, inject_primitives = false)
```

Comment:
I don't think `wants_doc` should have a different default value from `::semantic`, as it is entirely possible we want to use `top_level_semantic` in other places